pkg/backup/backup_new_test.go

```diff
@@ -1254,6 +1254,382 @@ func TestBackupActionAdditionalItems(t *testing.T) {
 	}
 }
 
+// volumeSnapshotterGetter is a simple implementation of the VolumeSnapshotterGetter
+// interface that returns velero.VolumeSnapshotters from a map if they exist.
+type volumeSnapshotterGetter map[string]velero.VolumeSnapshotter
+
+func (vsg volumeSnapshotterGetter) GetVolumeSnapshotter(name string) (velero.VolumeSnapshotter, error) {
+	snapshotter, ok := vsg[name]
+	if !ok {
+		return nil, errors.New("volume snapshotter not found")
+	}
+
+	return snapshotter, nil
+}
+
+func int64Ptr(val int) *int64 {
+	i := int64(val)
+	return &i
+}
+
+type volumeIdentifier struct {
+	volumeID string
+	volumeAZ string
+}
+
+type volumeInfo struct {
+	volumeType  string
+	iops        *int64
+	snapshotErr bool
+}
+
+// fakeVolumeSnapshotter is a test fake for the velero.VolumeSnapshotter interface.
+type fakeVolumeSnapshotter struct {
+	// PVVolumeNames is a map from PV name to volume ID, used as the basis
+	// for the GetVolumeID method.
+	PVVolumeNames map[string]string
+
+	// Volumes is a map from volume identifier (volume ID + AZ) to a struct
+	// of volume info, used for the GetVolumeInfo and CreateSnapshot methods.
+	Volumes map[volumeIdentifier]*volumeInfo
+}
+
+// WithVolume is a test helper for registering persistent volumes that the
+// fakeVolumeSnapshotter should handle.
+func (vs *fakeVolumeSnapshotter) WithVolume(pvName, id, az, volumeType string, iops int, snapshotErr bool) *fakeVolumeSnapshotter {
+	if vs.PVVolumeNames == nil {
+		vs.PVVolumeNames = make(map[string]string)
+	}
+	vs.PVVolumeNames[pvName] = id
+
+	if vs.Volumes == nil {
+		vs.Volumes = make(map[volumeIdentifier]*volumeInfo)
+	}
+
+	identifier := volumeIdentifier{
```

Comment:
would it be simpler if volumeIdentifier was just a string concatenation of volumeID and volumeAZ? It doesn't look like we're accessing the volumeID or volumeAZ values from the volumeIdentifier struct anywhere (though I'm probably missing something)

Reply:
That is an option, yeah. I often use structs instead of string concats in this kind of scenario to avoid the general issue where, for example, "a" + "bc" gives the same result as "ab" + "c". Shouldn't be a big issue here with a reasonable separator in the middle, e.g. `||`.

Reply:
Good point, yeah a separator would help there. Just a thought though, the struct is fine, just thought it might save a few lines over having to initialise the struct.